driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTestModifications.java

```diff
@@ -372,32 +366,24 @@ private TestApplicator(
                 final TestDef testDef,
                 final List<Modifier> modifiersToApply) {
             this.testDef = testDef;
-            this.shouldSkip = false;
-            this.reasonToApply = null;
             this.modifiersToApply = modifiersToApply;
         }
 
         private TestApplicator(
```

Comment:
Should we remove this ctor and force the caller to provide the modifiers (if any)? One could accidentally invoke this ctor and have all tests skipped...

Reply:
Good point, I cleaned this up to have one constructor, and this also makes it clearer what is happening in the methods that call the constructor.